Library/HUMSlider.m

```diff
@@ -109,25 +164,195 @@ - (void)nukeOldTicks
     [self layoutIfNeeded];
 }
 
+- (void)refreshView {
+    [self setupTicks];
+}
+
 - (void)setupTicks
 {
+    if ([self areCustomTicksSetupAndNonNull]) {
+        [self cleanupAfterEvenlySpacedTicks];
+        [self setupCustomTickViews];
+        [self setupTicksAutoLayoutCustomWidths];
+        if (!_enableTicksTransparencyOnIdle) {
+            [self animateAllTicksInCustomWidths:YES];
+        }
+    }
+    else {
+        [self cleanupAfterCustomSpacedTicks];
+        [self setupSpacedTickViews];
+        [self setupTicksAutolayout];
+        if (!_enableTicksTransparencyOnIdle) {
+            [self animateAllTicksIn:YES];
+        }
+    }
+    
+
+}
+
+- (void)cleanupAfterEvenlySpacedTicks {
+    self.tickViews = [NSArray new]; // Make sure these get re-initialized
+    [self clearLayoutConstraintList:_leftTickRightConstraints];
+    [self clearLayoutConstraintList:_rightTickLeftConstraints];
+    [self clearLayoutConstraintList:_allTickBottomConstraints];
```

Comment:
For all these (and in the method below) you should be able to do `[self.allTickBottomConstraints removeAllObjects]` directly. [`removeAllObjects` docs](https://developer.apple.com/documentation/foundation/nsmutablearray/1410618-removeallobjects?language=objc)